packages/clarity-js/src/layout/animation.ts

```diff
@@ -0,0 +1,77 @@
+import { Event } from "@clarity-types/data";
+import { AnimationOperation, AnimationState } from "@clarity-types/layout";
+import { time } from "@src/core/time";
+import { shortid } from "@src/data/metadata";
+import encode from "@src/layout/encode";
+import { getId } from "@src/layout/dom";
+
+export let state: AnimationState[] = [];
+let animationPlay: () => void = null;
+let animationPause: () => void = null;
+let animationCancel: () => void = null;
+let animationFinish: () => void = null;
+
+export function start(): void {
+    reset();
+    overrideAnimationHelper(animationPlay, "play");
+    overrideAnimationHelper(animationPause, "pause");
+    overrideAnimationHelper(animationCancel, "cancel");
+    overrideAnimationHelper(animationFinish, "finish");
+}
+
+export function reset(): void {
+    state = [];
+}
+
+export function track(time: number, id: string, operation: AnimationOperation, keyFrames?: string, timing?: string, targetId?: number, timeline?: string): void {
+    state.push({
+        time,
+        event: Event.Animation,
+        data: {
+            id,
+            operation,
+            keyFrames,
+            timing,
+            targetId,
+            timeline
+        }
+    });
+
+    encode(Event.Animation);
+}
+
+export function stop(): void {
+    state = [];
+    reset();
+}
+
+function overrideAnimationHelper(whereToStoreFunction: () => void, name: string) {
+    if (whereToStoreFunction === null) {
+      whereToStoreFunction = Animation.prototype[name];
+      Animation['clarity'] = true;
+      Animation.prototype[name] = function(): void {
+        if (!this['clarityAnimationName']) {
```

Comment:
>clarityAnimationName [](http://example.com/codeflow?start=19&length=20)

Would clarityAnimationId be better? Also think it would best to store this as a const outside since the same string is used in many places.